src/main/src/test/java/org/geoserver/catalog/impl/CatalogBuilderTest.java

```diff
@@ -323,6 +323,28 @@ public void testNativeBoundsDefensiveCopy() throws Exception {
         }
     }
 
+    @Test
+    public void testBoundingBoxGeneration() throws Exception {
+        Catalog cat = getCatalog();
+        CatalogBuilder cb = new CatalogBuilder(cat);
+        cb.setStore(cat.getCoverageStoreByName(MockData.TASMANIA_DEM.getLocalPart()));
+        CoverageInfo ci = cb.buildCoverage();
+
+        ReferencedEnvelope bbox = ci.getNativeBoundingBox();
+        assertNotNull(bbox.getCoordinateReferenceSystem());
+
+        // simulate missing srs
+        ci.setNativeBoundingBox(ReferencedEnvelope.create(bbox, null));
+        assertEquals(ci.getNativeCRS(), ci.boundingBox().getCoordinateReferenceSystem());
+
+        // simulate everything being missing
+        ci.setNativeBoundingBox(null);
+        ci.setSRS(null);
+
+        // this will attempt to back-project form LatLon
+        assertNull(ci.boundingBox());
```

Comment:
Hmm... an exception might be more useful here, we'd just risk getting a NPE in another part of the code and have trouble tracking it down?

Reply:
The boundingBox() method is already returning null if it cannot figure things out so I figured that this would be okay?